oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/HttpSessionOAuth2AuthorizationRequestRepository.java

```diff
@@ -113,14 +137,70 @@ private String getStateParameter(HttpServletRequest request) {
 	 * @return a non-null and mutable map of {@link OAuth2AuthorizationRequest#getState()}
 	 * to an {@link OAuth2AuthorizationRequest}.
 	 */
-	private Map<String, OAuth2AuthorizationRequest> getAuthorizationRequests(HttpServletRequest request) {
+	private Map<String, OAuth2AuthorizationRequestReference> getAuthorizationRequests(HttpServletRequest request) {
 		HttpSession session = request.getSession(false);
-		Map<String, OAuth2AuthorizationRequest> authorizationRequests = (session != null)
-				? (Map<String, OAuth2AuthorizationRequest>) session.getAttribute(this.sessionAttributeName) : null;
+		Map<String, OAuth2AuthorizationRequestReference> authorizationRequests = (session != null)
+				? (Map<String, OAuth2AuthorizationRequestReference>) session.getAttribute(this.sessionAttributeName)
+				: null;
 		if (authorizationRequests == null) {
 			return new HashMap<>();
 		}
+		// remove expired entries
+		authorizationRequests.entrySet().removeIf((entry) -> entry.getValue().expiresAt.isBefore(this.clock.instant()));
 		return authorizationRequests;
 	}
 
+	/**
+	 * Sets the {@link Clock} used in {@link Instant#now(Clock)} when setting the instant
+	 * created for {@link OAuth2AuthorizationRequest}.
+	 * @param clock the clock
+	 * @since 5.5
+	 */
+	void setClock(Clock clock) {
```

Comment:
Please remove this as it's not needed. The clock will always be in sync since this implementation operates independently within it's own application instance. Clock and clock skew is needed when components are interacting in a distributed way.

Reply:
This method is package private, so nothing other than Spring itself should be using it.

Being able to set the clock is important for testing. See https://github.com/candrews/spring-security/blob/9b1a85d992f3ac0d725f67ac03ee6d9f44659b21/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/HttpSessionOAuth2AuthorizationRequestRepositoryTests.java#L249

Is there another approach that should be taken to make the clock available to be set by tests?